server/sandbox_run_linux.go

```diff
@@ -826,76 +848,67 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 		g.AddAnnotation(fmt.Sprintf("%s.%d", annotations.IP, idx), ip)
 	}
 	sb.AddIPs(ips)
-	sb.SetNamespaceOptions(securityContext.GetNamespaceOptions())
 
-	spp := securityContext.GetSeccompProfilePath()
-	g.AddAnnotation(annotations.SeccompProfilePath, spp)
-	sb.SetSeccompProfilePath(spp)
-	if !privileged {
-		if err := s.setupSeccomp(ctx, &g, spp); err != nil {
-			return nil, err
-		}
-	}
-
-	err = g.SaveToFile(filepath.Join(podContainer.Dir, "config.json"), saveOptions)
-	if err != nil {
+	if err = g.SaveToFile(filepath.Join(podContainer.Dir, "config.json"), saveOptions); err != nil {
 		return nil, fmt.Errorf("failed to save template configuration for pod sandbox %s(%s): %v", sb.Name(), sbox.ID(), err)
 	}
 	if err = g.SaveToFile(filepath.Join(podContainer.RunDir, "config.json"), saveOptions); err != nil {
 		return nil, fmt.Errorf("failed to write runtime configuration for pod sandbox %s(%s): %v", sb.Name(), sbox.ID(), err)
 	}
 
-	s.addInfraContainer(container)
-	defer func() {
-		if retErr != nil {
-			log.Infof(ctx, "runSandbox: removing infra container %s", container.ID())
-			s.removeInfraContainer(container)
-		}
-	}()
+	if !container.Spoofed() {
+		s.addInfraContainer(container)
+		defer func() {
+			if retErr != nil {
+				log.Infof(ctx, "runSandbox: removing infra container %s", container.ID())
+				s.removeInfraContainer(container)
+			}
+		}()
 
-	if sandboxIDMappings != nil {
-		rootPair := sandboxIDMappings.RootPair()
-		for _, path := range pathsToChown {
-			if err := makeAccessible(path, rootPair.UID, rootPair.GID, true); err != nil {
-				return nil, errors.Wrapf(err, "cannot chown %s to %d:%d", path, rootPair.UID, rootPair.GID)
+		if sandboxIDMappings != nil {
+			rootPair := sandboxIDMappings.RootPair()
+			for _, path := range pathsToChown {
+				if err := makeAccessible(path, rootPair.UID, rootPair.GID, true); err != nil {
+					return nil, errors.Wrapf(err, "cannot chown %s to %d:%d", path, rootPair.UID, rootPair.GID)
+				}
+			}
+			if err := makeMountsAccessible(rootPair.UID, rootPair.GID, g.Config.Mounts); err != nil {
+				return nil, err
 			}
 		}
-		if err := makeMountsAccessible(rootPair.UID, rootPair.GID, g.Config.Mounts); err != nil {
+
+		if err := s.createContainerPlatform(container, sb.CgroupParent(), sandboxIDMappings); err != nil {
 			return nil, err
 		}
-	}
-
-	if err := s.createContainerPlatform(container, sb.CgroupParent(), sandboxIDMappings); err != nil {
-		return nil, err
-	}
 
-	if err := s.Runtime().StartContainer(container); err != nil {
-		return nil, err
-	}
+		if err := s.Runtime().StartContainer(container); err != nil {
+			return nil, err
+		}
```

Comment:
What caused this whitespace change? (I can't see it from github if it is correct or not)

Reply:
this does look weird, but I believe it's fixed in a later commit (which I will be pushing a version that's been squashed soon)